### DISCUSSION TITLE
Video sobre como contribuir al repo usando fork

### DISCUSSION THREAD

**Discussion by sebasvelasco353** [sebasvelasco353/FDP-2022_01#4] [Announcements]
Hola, les dejo el [link](https://youtu.be/y2G9ePYV3C8) al video en youtube donde les explico como subir sus cambios a su repo y despues abrir un pull request.

si tienen alguna duda pueden preguntarme usando la discusion de Q&A.

**Comment by Darksea1220** [comment]
Hola profe, soy Douglas Osorio, profe, cuando le doy el git remote -v me sale "fatal: not a git repository (or any of the parent directories): .git"

**Reply by sebasvelasco353** [reply]
Hola Douglas, eso pasa porque no estás dentro de la carpeta del proyecto que clonaste en tu computador.
Puedes usar el comando pwd para saber en qué directorio estás y usando cd puedes navegar a la carpeta del repo.

Psd: ../ significa subir una carpeta y ./ significa en la carpeta en donde estoy en este momento.

**Reply by Darksea1220** [reply]
profe, no sé en qué carpeta está eso y con la que creo que es, uso el cd y me sale " No such file or directory"

**Reply by sebasvelasco353** [reply]
Ya clonaste el repo a tu computador? Podes clonarlo copiando el url que aparece en tu repo en github, para eso dale click al botón verde que dice clonar  y copias la url que aparece ahí.

Después abrís la consola de git en Windows o la terminal en mac y navegas a donde queres tener el proyecto, supongamos que a los documentos entonces cd Documents y cuando ya estás ahí le das git clone y el url que copiaste del repo.

Esto va a copiar el repo a tu compu y le das en la consola o terminal cd y el nombre de la carpeta (FDP_ etc).

Ya una vez estando adentro de la carpeta del repo puedes hacer todo lo del video normal

**Reply by sebasvelasco353** [reply]
Tambien puedes usar pwd en la terminal y te va a decir en qué carpeta estás, puede ser que no estes en la correcta, otra opción es usar ls y te va a decir que archivos y carpetas están donde te encuentras en ese momento y así verificar si la carpeta que buscas está ahí o si tiene otro nombre.

**Reply by Darksea1220** [reply]
listo profe, ya lo pude actualizar, pero ahora cuando voy a subir mi commit, le doy el code . y me dice 
bash: code: command not found

**Reply by sebasvelasco353** [reply]
En ese caso puedes meterte a vscode y presionar ctrl+p o cmd+p y eso te va a abrir un drop down con posibles comandos, ahí buscas shell command: install code command, le das a ese y el te va a dejar el comando code listo para usarlo.

Después de eso ya puedes volver a usar code . Para abrir tu editor.

**Reply by Darksea1220** [reply]
en el drop down no aparece ningún shell command: install code command, los comandos que me salen con code son:

- Code: check for updates
- Code: show release notes
- clear profile code lenses
- Help: sing up for the vs code newsletter
- toggle source code in disassembli view
por shell no sale nada y por command salen

- terminal: select to previous command
- show emmet commands
- show all commands
- clear command history
- Terminal: run recent command
- Terminal: scroll to next command
- Terminal: scroll to previous command
- Terminal: select to next command

**Comment by Pipelenisf** [comment]
Hola profe, cuando intento hacer git push me aparece esto:
 ```
git push
remote: Permission to sebasvelasco353/FDP-2022_01.git denied to Pipelenisf.
fatal: unable to access 'https://github.com/sebasvelasco353/FDP-2022_01.git/': The requested URL returned error: 403
```

**Reply by sebasvelasco353** [reply]
Hola pipe, probablemente estás tratando de hacer push al repo de la clase y no al tuyo, El commit debes subirlo a tu repo y desde ahí abrir el pull request.

Puedes usar git remote -v para revisar que origin sea tu repositorio y upstream sea el de la clase, después de eso haces git push origin y debería funcionar.

**Comment by Kathereyes19** [comment]
Hola profe, cuando intento hacer git push me aparece esto. Que debo hacer?
<img width="615" alt="Captura de Pantalla 2022-02-03 a la(s) 11 15 13 p  m" src="https://user-images.githubusercontent.com/98465512/152470837-8172cf0e-a066-42e0-a774-50c43207ed86.png">

**Reply by sebasvelasco353** [reply]
Eso pasa porque tienes cosas nuevas en el repo que no tienes en tu local, primero tienes que hacer `git pull` para traer los cambios qué hay en el remoto y después de eso haces un commit y git push.

**Comment by sophienunez** [comment]
Profe, no logré subir los archivos por un error 403 que me aparecía entonces como alternativa subí la página a la web y a un repositorio que creé. No sé si mientras tanto eso pueda servir, ahí están los archivos html y css. En clase ya te mostraré bien cuál era el error que no me dejó subir los archivos como tú dijiste que lo hiciéramos

link del repositorio: https://github.com/sophienunez/gaspard-and-lisa.git
link de la página: https://sophienunez.github.io/gaspard-and-lisa/
